src/main/java/ch/wisv/toornament/concepts/Disciplines.java

```diff
@@ -15,13 +17,18 @@ public Disciplines(ToornamentClient client) {
         super(client);
     }
 
-    public List<Discipline> getDisciplines() {
+    public List<Discipline> getDisciplines() throws IOException {
         // TODO: PLACEHOLDER
-        return Collections.emptyList();
+        Request request = client.getAuthenticatedRequestBuilder().get().url("https://api.toornament.com/v1/disciplines").build();
+         String responseBody = client.executeRequest(request).body().string();
+        return mapper.readValue(responseBody, mapper.getTypeFactory().constructCollectionType(List.class, Discipline.class));
     }
 
-    public DisciplineDetails getDiscipline(String id) {
+    public DisciplineDetails getDiscipline(String id) throws IOException {
         // TODO: PLACEHOLDER
-        return new DisciplineDetails();
+        Request request = client.getAuthenticatedRequestBuilder().get().url("https://api.toornament.com/v1/disciplines"+ id ).build();
```

Comment:
I'm not sure how Jackson works, so I don't know how this should be done. I'm guessing the below, but won't commit it for now.
```
private class TeamSize {
        int min;
        int max;
        public TeamSize(){
            
        }
        public int getMin(){
            return this.min;
        }
        public int getMax(){
            return this.max;
        }
        
    }
```